### PR TITLE
Implement retrying logic in HttpCall

### PR DESCRIPTION
**Pull request by ignaloidas**

### CODE REVIEW

File: b_lambda_layer_common_test/util/test_http_call.py [hunk 1]
```diff
@@ -34,9 +38,34 @@ def test_FUNC_call_WITH_not_existing_endpoint_EXPECT_not_found() -> None:
...
+    :return: No return.
+    """
+    with pook.use():
+        mock = pook.get("https://unreachable.com").error(HTTPError()).times(5)
+        with pytest.raises(NotReachedError):
+            start = time.time()
+            HttpCall.call('GET', 'https://unreachable.com')
+        end = time.time()
+        time_taken = end - start
+        assert 7.5 < time_taken < 8.5
```
Comment (laimonassutkus) [inline b_lambda_layer_common_test/util/test_http_call.py:71]: Please add a comment why you expect "time_taken" to be in this interval.

**Review by laimonassutkus — CHANGES_REQUESTED** [review]
There is a lot to change. Can not approve.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/util/http_call.py [hunk 2]
```diff
@@ -43,22 +44,37 @@ def call(method, url, fields=None, headers=None, **urlopen_kw) -> HTTPResponse:
...
-                url=url,
-                fields=fields,
-                headers=headers,
-                **urlopen_kw
-            )
-        except HTTPError as ex:
-            raise NotReachedError(str(ex))
-
-        if 400 <= response.status <= 599:
+        wait_length = 0.249 # a bit less than 1/4 to avoid floating point problems
```
[the remark this answers is not in the document]
Comment (ignaloidas) [reply b_lambda_layer_common/source/python/b_lambda_layer_common/util/http_call.py:47]: It is not a constant, it gets modified by by code.

File: b_lambda_layer_common/source/python/b_lambda_layer_common/util/http_call.py [hunk 3]
```diff
@@ -19,6 +20,14 @@
 
 
 class HttpCall:
+    MAXIMUM_TOTAL_WAIT = 8
+    WAIT_EXPONENTIATION_FACTOR = 2
+
+    MAXIMUM_SINGLE_WAIT = MAXIMUM_TOTAL_WAIT / (WAIT_EXPONENTIATION_FACTOR / (WAIT_EXPONENTIATION_FACTOR -1))
```
Comment (laimonassutkus) [inline b_lambda_layer_common/source/python/b_lambda_layer_common/util/http_call.py:26]: Please think of an easier implementation.

**Review by laimonassutkus — CHANGES_REQUESTED** [review]

**Review by laimonassutkus — APPROVED** [review]

**Review by laimonassutkus — APPROVED** [review]

**Pull request merged**